### PR TITLE
New Bootstrap-based UI

### LINKED ISSUE #180: Add a modern bootstrap-like UI
Using boostrap for the UI could help with customization and integration in other apps IMHO

### PR DESCRIPTION
**Pull request by gwasser**
Following an open source template (see LICENSE.3RDPARTY) that makes heavy use of Bootstrap and some jQuery, I freshened up the UI. It's mostly just some nicer styling, but I did move around a few elements.
- The dashboard provides only status update of open tickets, with the statistics moved to the statistics/reports page. 
- The ticket list window is now a full table with features like sort-by-column-on-click, really snazzy (don't thank me, thank the template for ideas!).
- Across whole site, plain links were updated to be CSS buttons to fit the theme better. I tried to stick to the convention of "green" is "safe" change, "blue" is a standard action, "yellow" is cautious one that makes changes to a ticket, "red" is dangerous (primarily for deleting something).
- It also now uses Font Awesome, adding possibility for many more icons/glyphs. I added icons to many of the buttons already. Maybe more is possible.
- The python backend was slightly adapted in a couple of places to support the new UI, but no major features were added/removed (was mostly adding a variable to the dictionary fed to templates).

I'm sure more UI work can be done by someone more expert than I, but I think this a good start. I wanted to have this in place for a personal project of mine that needs a better UI to build on.

This might also partially/completely resolve #180 .

Thanks for allowing me to contribute! Appreciate any comments you may have.

### CODE REVIEW

**Comment by rossp** [comment]
Thanks for this comprehensive update! is there any chance you could share a couple of screenshots with us? I am going to be unable to properly review this until mid-October at the earliest due to travel but  would love to see how it looks, and it'd be great if others can take a look too.

**Comment by gwasser** [comment]
Certainly! Please find some screenshots below. Like I said, could probably still use some work but I think it looks a little less plain now.

Not a problem to wait until October or later. I didn't style the new ticket form and a couple others since they use bootstrap-form, but based on #313 maybe it's best to remove bootstrap-form and hardcode the forms to our liking (with correct styling). Since you can't review until October, is it ok to push a few more changes to this branch to maybe hardcode the forms and finish conversion?

screenshot_20160913_004208
screenshot_20160913_004440
screenshot_20160913_004522
screenshot_20160913_004807
screenshot_20160913_004825
screenshot_20160913_004851
screenshot_20160913_004925
screenshot_20160913_005011
screenshot_20160913_005130
screenshot_20160913_005205
screenshot_20160913_005227

**Comment by fchevitarese** [comment]
This new template is very nice! When this will be merged? =D

**Comment by gwasser** [comment]
There's new maintainers and I think we're working out the full process on new merges still. But I hope soon!

Maintainers: I maybe think since its a large visual overhaul that this commit be a good candidate to start a 0.2 branch. I suggest we branch 0.1 off of master and tag it (so the "classic" version is available easily to anyone that wants it), then master becomes 0.2 dev. Any thoughts?

**Comment by reduxionist** [comment]
@gwasser yes that's a good workflow IMHO

@fchevitarese I'm currently wrapping up three-weeks of travel in North America, but will be flying back home to Asia on Monday. Since I have a ton of backlog to clear first, I won't be able to do any large-scale code-reviews here until the following weekend, so the earliest merge available will probably be @gwasser's 0.2-dev branch.

File: helpdesk/templates/helpdesk/include/tickets.html [hunk 1]
```diff
@@ -1,21 +1,46 @@
...
+                                            <th>{% trans "Title" %}</th>
+                                            <th>{% trans "Queue" %}</th>
+                                            <th>{% trans "Status" %}</th>
+                                            <th>{% trans "Last Update" %}</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        {% for ticket in ticket_list %}
+                                        <tr class="{{ ticket.get_priority_css_class }}">
+                                            <td><a href='{{ ticket.get_absolute_url }}'>{{ ticket.ticket }}</td>
```
Comment (reduxionist) [inline helpdesk/templates/helpdesk/include/tickets.html]: Missing close anchor tag.

File: helpdesk/templates/helpdesk/include/unassigned.html [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+                                            <th>{% trans "Title" %}</th>
+                                            <th>{% trans "Queue" %}</th>
+                                            <th>{% trans "Created" %}</th>
+                                            <th>&nbsp;</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        {% for ticket in unassigned_tickets %}
+                                        <tr class="{{ ticket.get_priority_css_class }}">
+                                            <td><a href='{{ ticket.get_absolute_url }}'>{{ ticket.ticket }}</td>
```
Comment (reduxionist) [inline helpdesk/templates/helpdesk/include/unassigned.html]: Missing close anchor tag here too.

File: helpdesk/templates/helpdesk/kb_index.html [hunk 3]
```diff
@@ -5,17 +5,23 @@
...
+    <div class="col-lg-4">
+        <div class="panel panel-primary">
+            <div class="panel-heading">
+                <h3>{{ category.title }}</h3>
+            </div>
+            <div class="panel-body">
+                <p>{{ category.description }}</p>
+            </div>
+            <div class="panel-footer">
+                View <a href='{{ category.get_absolute_url }}'>{{ category.title }} articles <i class="fa fa-arrow-right"></i></a>
```
Comment (reduxionist) [inline helpdesk/templates/helpdesk/kb_index.html]: Need a blocktrans tag here to wrap this up into a translator-friendly-style "View (%s) articles".

**Comment by reduxionist** [comment]
I still haven't worked out a good way to do an in-line review on a PR this size via GitHub's web interface, so in the interest of getting this branch-motivating PR merged before there's any more updating required, I'm dismissing my i18n and future-proofing nitpicks and only waiting on the build-breaking url-namespacing fix before merge.

**Comment by gwasser** [comment]
OK should be good for merging now. I went through my email notifications and made the improvements you suggested, but possibly I missed a few. Probably good for later PRs/cleanups. I tested it locally and everything seems good.

**Comment by alexbarcelo** [comment]
:tada: Wow! Awesome PR :D

Can't wait to test it. I am just now adding it into the stage deployment of my non-profit organization, to test and break things a little bit ;)

Great work @gwasser

**Comment by gwasser** [comment]
Thanks! I appreciate @reduxionist time reviewing such a large PR. It's probably not perfect (I'm not a web GUI expert) but I think a good step forward.

File: helpdesk/templates/helpdesk/navigation.html [hunk 4]
```diff
@@ -9,35 +9,36 @@
...
+
                 {% endif %}
                 {% if user_saved_queries_ %}
                 <li class="headerlink dropdown"><a class="dropdown-toggle" data-toggle="dropdown" href="#"><i class="fa fa-filter fa-fw"></i> <span class="nav-text">{% trans "Saved Query" %} <b class="caret"></b></span></a>
                     <ul class="dropdown-menu">
                         {% for q in user_saved_queries_ %}
-                            <li><a href="{% url 'helpdesk_list' %}?saved_query={{ q.id }}">{{ q.title }}
+                            <li><a href="{% url 'helpdesk:list' %}?saved_query={{ q.id }}">{{ q.title }}
                                 {% if q.shared %}
                                     (Shared{% ifnotequal user q.user %} by {{ q.user.get_username }}{% endifnotequal %})
```
[the remark this answers is not in the document]
Comment (nickmccullum) [reply helpdesk/templates/helpdesk/navigation.html:43]: @reduxionist this broke the upgrade to Django 4, 6 years later...

**Pull request merged**